app/Commands/NewCommand.php

```diff
@@ -0,0 +1,474 @@
+<?php
+
+namespace App\Commands;
+
+use Illuminate\Filesystem;
+use Symfony\Component\Process\Process;
+use LaravelZero\Framework\Commands\Command;
+use Symfony\Component\Process\ExecutableFinder;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+
+class NewCommand extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'new
+                        {name : Name of the Laravel project}
+                        {--a|auth : Run make:auth}
+                        {--b|browser : Browser you want to open the project in}
+                        {--c|createdb= : Create a database; pass in sqlite or mysql}
+                        {--d|dev  : Choose the dev branch instead of master}
+                        {--e|editor= : Text editor to open the project in}
+                        {--l|link : Create a Valet link to the project directory}
+                        {--m|message= : Set the first commit message}
+                        {--y|node : Set to execute yarn or npm install}
+                        {--p|path= : Base path for the installation, otherwise CWD is used}';
+
+    protected $projectname = '';
+
+    protected $projectpath = '';
+
+    protected $projecturl = '';
+
+    protected $cwd = '';
+
+    protected $basepath = '';
+
+    protected $commitmessage = 'Initial commit.';
+
+    protected $editors_terminal = ['vim', 'vi', 'nano', 'pico', 'ed', 'emacs', 'nvim'];
+
+    protected $editors_gui = ['pstorm', 'subl', 'atom', 'textmate', 'geany'];
+
+    protected $editor = '';
+
+    protected $tools = [];
+
+    protected $dbtypes = ['sqlite', 'mysql'];
+
+    protected $os = null;
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Creates a new Laravel application';
+
+    /**
+     * Create a new command instance.
+     */
+    public function __construct()
+    {
+        parent::__construct();
+
+        $this->cwd = getcwd();
+        $this->basepath = $this->cwd;
+
+        $this->os = new \Tivie\OS\Detector();
+    }
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle(): void
+    {
+        $this->projectname = $this->argument('name');
+        $this->projecturl = 'http://'.$this->projectname.'.dev';
+
+        $this->getAvailableTools();
+        if (! $this->tools['laravel']) {
```

Comment:
This is great, thanks, and `hasTool()` is now implemented. Tokenizing (cache) was also implemented just in case the future warrants it and/or someone decides to steal `hasTool()` and use it for other things.